libsolidity/CMakeLists.txt

```diff
@@ -28,7 +28,7 @@ else()
 endif()
 
 add_library(solidity ${sources} ${headers})
-target_link_libraries(solidity PUBLIC evmasm devcore)
+target_link_libraries(solidity PUBLIC evmasm devcore ${Boost_FILESYSTEM_LIBRARY} ${Boost_SYSTEM_LIBRARY})
```

Comment:
If you use CMake 3.5+, you could use imported targets (read https://cmake.org/cmake/help/v3.5/module/FindBoost.html)

This would become `target_link_libraries(solidity PUBLIC evmasm devcore Boost::filesystem Boost::system)`.

Reply:
But we only require user to use cmake 3.0.0+ now? ( https://github.com/ethereum/solidity/blob/develop/CMakeLists.txt#L1 )

Reply:
Yes, that's true. But in case you are not testing old CMake version regularly it might be broken already.

Reply:
What actually worries me here is that `libsolidity`should not use boost::filesystem.

Reply:
Ah ok, it does use it. Hm, perhaps we should change that.

Reply:
Hm, devcore links against boost::filesystem and boost::system - shouldn't that be enough?

Reply:
Maybe it is because we use `PRIVATE` keyword to link devcore? (https://github.com/ethereum/solidity/blob/develop/libdevcore/CMakeLists.txt#L5)
So `boost::filesystem` and `boost::system` are not made part of "link interface", not propagating to `libsolidity`.

Reply:
reference: https://cmake.org/cmake/help/v3.11/command/target_link_libraries.html#libraries-for-a-target-and-or-its-dependents

Reply:
That is probably the case. PRIVATE dependencies are propagated for static libs (because of the nature of the static libs), but probably not for shared libs. This is a bit messy, so I would go with a simple fix like this one that works.